cpp/include/cudf/strings/copying.hpp

```diff
@@ -48,14 +48,14 @@ namespace detail
  * @param stream CUDA stream to use kernels in this method.
  * @param mr Resource for allocating device memory.
  * @return New strings column of size (end-start)/step.
- *---------------------------------------------------------------------------**/
-std::unique_ptr<cudf::column> slice( strings_column_view strings,
+ */
+std::unique_ptr<cudf::column> slice( strings_column_view const& strings,
                                      size_type start, size_type end=-1,
                                      size_type step=1,
                                      cudaStream_t stream=0,
```

Comment:
Maybe swap the order of stream and mr for consistency with other APIs

Reply:
This would require changing code that currently calls this which is not related to this PR. 
I would rather do a separate PR to fix this.